pkg/mempool/custom_proposal_handler_test.go

```diff
@@ -0,0 +1,393 @@
+		// test 3
+		{tx: buildMsg(s.T(), txConfig, []byte(`9`), [][]byte{secret3, secret4}, []uint64{3, 3}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`1052345678910`), [][]byte{secret1, secret2}, []uint64{4, 4}), priority: 8},
+		{tx: buildMsg(s.T(), txConfig, []byte(`11`), [][]byte{secret1, secret2}, []uint64{5, 5}), priority: 8},
+		// test 4
+		{tx: buildMsg(s.T(), txConfig, []byte(`1252345678910`), [][]byte{secret1}, []uint64{3}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`13`), [][]byte{secret1}, []uint64{5}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`14`), [][]byte{secret1}, []uint64{6}), priority: 8},
+		// test 5, (same as test 1, but ethermint txs)
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`0`), [][]byte{secret1}, []uint64{1}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`12345678910`), [][]byte{secret1}, []uint64{2}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`22`), [][]byte{secret1}, []uint64{3}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`32`), [][]byte{secret2}, []uint64{1}), priority: 8},
+		// test 6, (same as test 4, but ethermint txs)
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`1252345678910`), [][]byte{secret1}, []uint64{3}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`13`), [][]byte{secret1}, []uint64{5}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`14`), [][]byte{secret1}, []uint64{6}), priority: 8},
+		// large cosmos tx to test with eth txs
+		{
+			tx: buildMsg(
+				s.T(),
+				txConfig,
+				[]byte(`1252345678910125234567891012523456789101252345678910125234567891012523456789101252345678910125234567891012523456789101252345678912343`),
+				[][]byte{secret2},
+				[]uint64{1},
+			),
+			priority: 8,
+		},
+	}
+
+	for i := range testTxs {
+		bz, err := txConfig.TxEncoder()(testTxs[i].tx)
+		s.Require().NoError(err)
+		testTxs[i].bz = bz
+		testTxs[i].size = int(cmttypes.ComputeProtoSizeForTxs([]cmttypes.Tx{bz}))
+	}
+
+	s.Require().Equal(testTxs[0].size, 111)
+	s.Require().Equal(testTxs[1].size, 121)
+	s.Require().Equal(testTxs[2].size, 112)
+	s.Require().Equal(testTxs[3].size, 112)
+	s.Require().Equal(testTxs[4].size, 195)
+	s.Require().Equal(testTxs[5].size, 205)
+	s.Require().Equal(testTxs[6].size, 196)
+	s.Require().Equal(testTxs[7].size, 196)
+	s.Require().Equal(testTxs[8].size, 196)
+
+	// ethermint txs
+	s.Require().Equal(testTxs[15].size, 247)
+	s.Require().Equal(testTxs[16].size, 257)
+	s.Require().Equal(testTxs[17].size, 248)
+	s.Require().Equal(testTxs[18].size, 248)
+
+	s.Require().Equal(testTxs[19].size, 259)
+	s.Require().Equal(testTxs[20].size, 248)
+	s.Require().Equal(testTxs[21].size, 248)
+
+	s.Require().Equal(testTxs[22].size, 248)
+
+	testCases := map[string]struct {
+		ctx         sdk.Context
+		txInputs    []testTx
+		req         abci.RequestPrepareProposal
+		handler     sdk.PrepareProposalHandler
+		expectedTxs []int
+	}{
```

Comment:
Can we add the two following test cases
-  Tx1,Tx2,Tx3  :  Where Tx2 and Tx3 have the same nonce , but Tx2 is larger than maxbytes . 
 The resultant response would be Tx1 and Tx3 
 
 - All valid transactions but only a few get added because of maxbytes limit

Reply:
- `skip same-sender non-sequential sequence and then add others txs` has a different sender for `testTxs[3]` 
- makes sense